### PR TITLE
Fixed Dense method and a new MKLPardiso for LODF evaluation

### PR DESCRIPTION
**Pull request by alefcastelli**
Significantly improved the Dense method for LODF, added MKLPardiso too

### CODE REVIEW

File: src/lodf_calculations.jl [hunk 1]
```diff
@@ -109,7 +109,7 @@ function _calculate_LODF_matrix_KLU(
     return lodf_t
 end
 
-# ! temp for evaluation
+# ! used only in testing, can be removed in the future
```
Comment (jd-lara) [inline src/lodf_calculations.jl:112]: What's the point of leaving this method here? If it isn't going to be part of the standard release is better to remove it at once.

**Review by jd-lara — CHANGES_REQUESTED** [review]
Let's remove the additional method that has the "for testing" comment.

File: test/test_lodf.jl [hunk 2]
```diff
@@ -63,22 +67,21 @@
...
-    # test if error is thrown in case `MKLPardiso` is chosen as a linera solver
+    # test if error is thrown in case `tol` is defined in PTDF
+    P5 = PTDF(sys5; tol = 1e-3)
     test_value = false
     try
-        lodf = LODF(sys5; linear_solver = "MKLPardiso")
+        L5NS_from_ptdf = LODF(A, P5)
     catch err
         if err isa ErrorException
             test_value = true
```
Comment (jd-lara) [inline test/test_lodf.jl:87]:
```julia
@test_throws ErrorException LODF(A, P5; linear_solver = "XXX")
```

File: test/test_lodf.jl [hunk 3]
```diff
@@ -63,22 +67,21 @@
...
     test_value = false
     try
-        L5NS_from_ptdf = LODF(A, P5)
+        L5NS_from_ba_aba = LODF(A, P5; linear_solver = "XXX")
     catch err
         if err isa ErrorException
             test_value = true
         end
     end
     @test test_value
```
[the remark this answers is not in the document]
Comment (alefcastelli) [reply test/test_lodf.jl:78]: ok

**Pull request merged**